File: 0035-search-insert-position/0035-search-insert-position.java
class Solution {
    public int searchInsert(int[] nums, int target) {
          int low=0;
        int high= nums.length-1;
        return binarySearch(nums,target,low,high);
        
    }
    public int binarySearch(int[] nums, int target,int low,int high){
        while(low<=high){
            int mid=(low+high)/2;
            if(nums[mid]==target){
                return mid;
            }
            else if(target>nums[mid]){
                low=mid+1;
            }
            else{
                high=mid-1;
            }
        }
        return low;
    }

}

File: 0035-search-insert-position/0035-search-insert-position.py
class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        n=len(nums)
        low =0
        high = n-1
        while low<= high:
            mid = low + (high-low)//2
            if nums[mid]==target:
                ans=mid
                return ans
            elif nums[mid]<target:
                low=mid+1
            else:
                high=mid-1
        return low
    
        